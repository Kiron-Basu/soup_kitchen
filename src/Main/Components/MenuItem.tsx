import React, { useState } from "react";
import { FoodItem } from "../Enums/FoodItems";
import "../Styles/MenuItem.scss";
import AddBoxIcon from "@material-ui/icons/AddBox";
import IndeterminateCheckBoxBoxIcon from "@material-ui/icons/IndeterminateCheckBox";
import {
  Button,
  Typography,
  Card,
  CardMedia,
  CardContent,
  CardHeader,
} from "@material-ui/core";
import bread from "../Images/bread.jpg";
import milk from "../Images/milk.jpg";
import cheese from "../Images/cheese.jpg";
import butter from "../Images/butter.jpg";
import { capitalise } from "../Utilities/InvoiceUtilities";

interface IMenuItem {
  addToCart: (foodItem: FoodItem, quantityState: number) => void;
  label: string;
}

export const MenuItem: React.FC<IMenuItem> = ({ addToCart, label }) => {
  const [quantityState, setQuantityState] = useState(0);
  const [updatedState, setUpdatedState] = useState(false);

  const handleAddToCart = () => {
    addToCart(foodItem, quantityState);
    setQuantityState(0);
    setUpdatedState(true);
    setTimeout(() => setUpdatedState(false), 1500);
  };

  const handleIncrease = () => {
    setQuantityState(quantityState + 1);
  };
  const handleDecrease = () => {
    if (quantityState > 0) {
      setQuantityState(quantityState - 1);
    }
  };
  const foodItem: FoodItem = FoodItem[label as keyof typeof FoodItem];
  return (
    <Card className="menuItem">
      <CardHeader title={capitalise(label)}></CardHeader>
      {/* <CardMedia component="img" image={label} title={label} /> */}
      {/* <img src={fp} alt="product" /> */}
      <div className={`menuItem__image menuItem__image--${label}`} />
      <div>
        <AddBoxIcon
          fontSize="large"
          color="action"
          onClick={handleIncrease}
          className="menuItem_quantityButtons"
        ></AddBoxIcon>
        <IndeterminateCheckBoxBoxIcon
          fontSize="large"
          color="action"
          onClick={handleDecrease}
          className="menuItem_quantityButtons"
        ></IndeterminateCheckBoxBoxIcon>
      </div>
      <CardContent>
        <Typography
          variant="body2"
          color="textSecondary"
          component="p"
        ></Typography>
        <Button
          onClick={handleAddToCart}
          variant="contained"
          color="primary"
          size="small"
        >
          Update cart
        </Button>
      </CardContent>
      {updatedState ? <p>Updated!</p> : <p>Quantity: {quantityState}</p>}
    </Card>
  );
};
