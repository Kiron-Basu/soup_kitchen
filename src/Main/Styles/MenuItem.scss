.menuItem {
  display: flex;
  flex-direction: column;
  //   grid-template-rows: repeat(5, 1fr);
  //   grid-gap: 5px;
  margin: 0 1vw 0 1vw;

  //   &_quantityButtons {
  //     justify-self: start;
  //     align-self: end;
  //     grid-row: 4 / 5;
  //   }
}
