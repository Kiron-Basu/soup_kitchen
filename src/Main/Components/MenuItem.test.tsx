import React from "react";
import { ShallowWrapper, shallow } from "enzyme";

import { MenuItem } from "./MenuItem";

describe("MenuItem tests", () => {
  const addToCart = jest.fn();

  test("Renders MenuItem correctly", () => {
    const component: ShallowWrapper = shallow(
      <MenuItem addToCart={addToCart} label={"test"} />
    );

    expect(component).toMatchSnapshot();
  });
});
