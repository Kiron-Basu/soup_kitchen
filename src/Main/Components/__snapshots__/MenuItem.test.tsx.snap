// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem tests Renders MenuItem correctly 1`] = `ShallowWrapper {}`;
